lib/galaxy/managers/workflows.py

```diff
@@ -924,6 +925,15 @@ def __connect_workflow_steps( self, steps, steps_by_external_id ):
 
             del step.temp_input_connections
 
+    def __set_default_label( self, module, state ):
+        """ Previously data input modules had a `name` attribute to rename individual steps. Here, this value is transferred
+        to the actual `label` attribute which is available for all module types, unique, and mapped to its own database column.
+        """
+        if not module.label and module.type in [ 'data_input', 'data_collection_input' ] and isinstance( state, dict ):
+            default_label = state.get( 'name' )
+            if str( default_label ).lower() != module.name.lower():
+                module.label = default_label
```

Comment:
I should definitely a comment here. The reason is that the `name` attribute was previously initialized with the `default_name`. In case of `data_input` this was `Input Dataset`, and for `data_collection_input` it was `Input Dataset Collection`. These default names are not-unique and should not be transferred to the `label` attribute since they would cause problems when a user loads the workflow and wants to save it after the names have been transferred to labels. In order to filter these default values, we are using the static `module.name` attribute which is set to `Input dataset` and `Input dataset collection`.